File: bindings/python/robot_bus/ros2_bridge/mappers/std_msgs/u_int8_multi_array.py
"""Generated mapper for `std_msgs/msg/UInt8MultiArray`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert
from robot_bus.ros2_bridge.mappers.std_msgs.multi_array_layout import multi_array_layout_to_bus, multi_array_layout_to_ros

def u_int8_multi_array_to_bus(msg):
    from robot_bus.std_msgs.msg.v1 import UInt8MultiArray as BusMsg

    bus = BusMsg()
    bus.layout.CopyFrom(multi_array_layout_to_bus(msg.layout))
    bus.data.extend(list(msg.data))
    return bus


def u_int8_multi_array_to_ros(bus):
    from std_msgs.msg import UInt8MultiArray as RosMsg

    out = RosMsg()
    out.layout = multi_array_layout_to_ros(bus.layout)
    out.data = list(bus.data)
    return out


class StdMsgsUInt8MultiArrayMapper:
    def ros_msg_type(self):
        from std_msgs.msg import UInt8MultiArray as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return u_int8_multi_array_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.std_msgs.msg.v1 import UInt8MultiArray as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return u_int8_multi_array_to_ros(bus)

File: src/ros2_bridge/mappers/std_msgs/uint8_multi_array.rs
//! Typed mapper for `std_msgs/msg/UInt8MultiArray`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn uint8_multi_array_to_bus(
    msg: ros_env::std_msgs::msg::UInt8MultiArray,
) -> crate::std_msgs::msg::v1::UInt8MultiArray {
    crate::std_msgs::msg::v1::UInt8MultiArray {
        layout: Some(
            crate::ros2_bridge::mappers::std_msgs::multi_array_layout::multi_array_layout_to_bus(
                msg.layout,
            ),
        ),
        data: crate::ros2_bridge::mappers::convert::u32_seq(msg.data),
    }
}

pub(crate) fn uint8_multi_array_to_ros(
    bus: crate::std_msgs::msg::v1::UInt8MultiArray,
) -> ros_env::std_msgs::msg::UInt8MultiArray {
    ros_env::std_msgs::msg::UInt8MultiArray {
        layout:
            crate::ros2_bridge::mappers::std_msgs::multi_array_layout::multi_array_layout_to_ros(
                bus.layout.unwrap_or_default(),
            ),
        data: crate::ros2_bridge::mappers::convert::FromU32Seq::from_u32_seq(bus.data),
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct StdMsgsUInt8MultiArrayMapper;

impl TypedTopicMapper for StdMsgsUInt8MultiArrayMapper {
    type Ros = ros_env::std_msgs::msg::UInt8MultiArray;
    type Bus = crate::std_msgs::msg::v1::UInt8MultiArray;

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(uint8_multi_array_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(uint8_multi_array_to_ros(msg))
    }
}
